src/api/ledger/parse/utils.js

```diff
@@ -62,8 +62,8 @@ function parseOutcome(tx: Object): ?Object {
     fee: utils.common.dropsToXrp(tx.Fee),
     balanceChanges: balanceChanges,
     orderbookChanges: orderbookChanges,
-    ledgerVersion: tx.ledger_index,
-    indexInLedger: tx.meta.TransactionIndex
+    ledgerVersion: tx.ledger_index || 0,
```

Comment:
Yes, I think if `outcome` is present, then `ledgerVersion` and `indexInLedger` should always be present.

> can unvalidated transactions be part of getTransactions response?

Yes